comment/views/comments.py

```diff
@@ -45,22 +57,41 @@ def form_valid(self, form):
         model_id = self.request.POST.get('model_id')
         model_object = get_model_obj(app_name, model_name, model_id)
         parent_id = self.request.POST.get('parent_id')
-        parent_comment = None
         if parent_id:
             parent_qs = Comment.objects.filter(id=parent_id)
             if parent_qs.exists():
                 parent_comment = parent_qs.first()
+        else:
+            parent_comment = None
```

Comment:
Please revert the code to the old state.
```suggestion
        parent_comment = None
        if parent_id:
            parent_qs = Comment.objects.filter(id=parent_id)
            if parent_qs.exists():
                parent_comment = parent_qs.first()
            
```

Reply:
In my opinion, it is better to define variables where they are used. While going through this logic, I was first confused as to why the variable `parent_comment` was given a value `None` before going I read the lines below this. The newer version seems more clear to me unless there is some other issue.

Reply:
I believe we should always initialise the variable and then update its value. what if `parent_qs.exists()` is false? The parent_comment will be undefined.

Reply:
If `parent_id` exists `parent_qs` would exist? Do i miss something here?

Reply:
This is not always true, what if the author of parent comment deleted it while another user is replying to it? The parent_id exists since is passed with comment form and parent_qs doesn't exists. To make it simple this code logic is not valid in other languages :)

Reply:
Yes, you are right.